pkg/auth/providers/azure/azure_provider.go

```diff
@@ -77,7 +77,7 @@ func (ap *azureProvider) RefetchGroupPrincipals(principalID string, secret strin
 	if err != nil {
 		return nil, err
 	}
-	azureClient, err := ap.newAzureClient(cfg, secret)
+	azureClient, err := ap.newAzureClientFromSecret(cfg, secret)
```

Comment:
This function seems to always use the old side code. We still need to be able to call this function (RefetchGroupPrincipals) in the nightly refresh if we are using the upgraded provider. 

In the current form, I would expect new side code to straight up not work during the nightly refresh, which would be bad.

Reply:
Good catch, updating to return the appropriate client based on the Graph endpoint - now it always returns the old client.

Reply:
Fixed - resolving.